File: idea/testData/hierarchy/overrides/javaMethodInInterface/main0.java
interface A {
    public void <caret>foo();
}

class B implements A {
    @Override
    public void foo() {

    }
}

class C implements T {
    @Override
    public void foo() {

    }
}

class D extends Z {
    @Override
    public void foo() {

    }
}

class S {

}

File: kotlin-ide/idea/testData/hierarchy/overrides/kotlinFunctionInClass/main0.kt
open class T {
    open fun <caret>foo() {

    }
}

open class X: T() {
    override fun foo() {

    }
}

open interface Y: T() {
    override fun foo() {

    }
}

open class Z: Y {
    override fun foo() {

    }
}

class SS {

}